debug: make toolbar retain x-percent on resize, avoid window control underflow

connor4312 opened the pull request:
Fixes #225551

<!-- Thank you for submitting a Pull Request. Please:
* Read our Pull Request guidelines:
  https://github.com/microsoft/vscode/wiki/How-to-Contribute#pull-requests
* Associate an issue with the Pull Request.
* Ensure that the code is up-to-date with the `main` branch.
* Include a description of the proposed changes and how to test them.
-->

connor4312 committed:
debug: make toolbar retain x-percent on resize, avoid window control underflow

Fixes #225551

Tyriar commented on src/vs/workbench/contrib/debug/browser/debugToolBar.ts:
```diff
@@ -62,6 +64,9 @@ export class DebugToolBar extends Themable implements IWorkbenchContribution {
 
 	private readonly trackPixelRatioListener = this._register(new MutableDisposable());
 
+	/** Note: changes to this setting require a restarts, so no need to listen to it. */
```
```suggestion
	/** Note: changes to this setting require a restart, so no need to listen to it. */
```

Tyriar commented on src/vs/workbench/contrib/debug/browser/debugToolBar.ts:
```diff
@@ -188,10 +193,18 @@ export class DebugToolBar extends Themable implements IWorkbenchContribution {
+		const nextFrameScheduler = this._register(new RunOnceScheduler(() => {
+			const thisWindow = dom.getWindow(this.layoutService.activeContainer);
+			this.setCoordinates(this.getCurrentXPercent() * thisWindow.innerWidth);
+		}, 30));
```
Next frame would be 0, why 30?

connor4312 replied:
I actually just wanna make some performance tweaks so this isn't necessary.

connor4312 committed:
improve toolbar performance, drag behavior

Use CSS transforms and avoid manual calculating the dimensions of the toolbar. This allows the toolbar to be moved around without causing layout changes and removes the need for the debounce added in the last commit.

Also improves the dragging logic to be based on mouse origin position instead of hardcoding the center of the drag handle as the origin.

connor4312 commented:
actually, with css calculations I can avoid needing to listen to resize at all!

connor4312 committed:
avoid listening to resize at all
